clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/api/KubernetesApiConverter.groovy

```diff
@@ -808,11 +808,19 @@ class KubernetesApiConverter {
         builder = builder.withNewRollingUpdate()
 
         if (rollingUpdate.maxSurge) {
-          builder = builder.withNewMaxSurge(rollingUpdate.maxSurge)
+          def maxSurge = rollingUpdate.maxSurge
```

Comment:
Makes sense.
There is apparently also a groovy `isInteger()` method you can use to avoid the try/catch technique entirely.